samcli/lib/telemetry/metrics.py

```diff
@@ -98,6 +99,17 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def sam_boto3_client(*args, **kwargs):
+    user_agent_extra = "aws-sam-{}".format(__version__)
+    if _telemetry_enabled():
+        installation_id = GlobalConfig().installation_id
+        user_agent_extra += '-'.join([user_agent_extra, installation_id])
+    return boto3.client(*args, config=Config(user_agent_extra=user_agent_extra), **kwargs)
```

Comment:
Thanks for this change, can you add a unit test here? to see what are the actual arguments boto3.client gets called with.